src/main/scala/engine/SimulationHistory.scala

```diff
@@ -27,6 +29,23 @@ object SimulationHistory{
     else KindsUtils.setAlleleDominance(mutation.geneKind.base)
   }
 
+  /** Introduce a new factor */
+  def introduceFactor(factor: Factor): Unit = {
+    getActualGeneration.environment.factors = factor :: getActualGeneration.environment.factors
+    if (factor.equals(DisturbingFactors.HIGH_FOOD) || factor.equals(DisturbingFactors.LIMITED_FOOD)
+        || factor.equals(DisturbingFactors.TOUGH_FOOD))
+      updateFoodView(getActualGeneration.environment.factors.filter(f => f != DisturbingFactors.WOLF &&
+        f != DisturbingFactors.HOSTILE_TEMPERATURE), getActualGeneration.environment.climate)
+
+  }
+
+  /** Remove a factor */
+  def removeFactor(factor: Factor): Unit = {
+    getActualGeneration.environment.factors = getActualGeneration.environment.factors.filter(_ != factor)
+    updateFoodView(getActualGeneration.environment.factors.filter(f => f != DisturbingFactors.WOLF &&
+      f != DisturbingFactors.HOSTILE_TEMPERATURE), getActualGeneration.environment.climate)
+  }
+
   /** Reset all the mutations added */
```

Comment:
Questi metodi li hai implementati seguendo la tua versione di testing (suppongo) del model, ma non si interfacciano per niente con il mio model. A parte questa nota, qui stai violando MVC perchè stai facendo fare delle operazioni sulla gui di visualizzazione del cibo quando in realtà non si ha consapevolezza di quello che sta succedendo dentro all'implementazione grafica.

Diverso è il discorso per la visualizzazione dei lupi e l'aggiornamento dei conigli, perchè questi due sono eventi scatenati direttamente dal model